### PR TITLE
Stubs for WebAssembly

### PR DESCRIPTION
**Pull request by neilalexander**
This should allow NATS Server to build under the `js/wasm` target.

/cc @nats-io/core

### CODE REVIEW

**Comment by derekcollison** [comment]
Should we be tracking this one still @neilalexander?

**Comment by neilalexander** [comment]
@derekcollison If you don't mind — this PR pretty much got NATS to build for WASM but I am not 100% sure of the consequences of stubbing some of these values.

**Comment by gedw99** [comment]
> 
@derekcollison @neilalexander 
So then needs built tags, so that the code can work in browser and not in browser right ? 
Then at compile time just use the flags appropriately. 

Here is an example of what i mean: https://github.com/gioui/gio-x/tree/main/explorer

keeps everything reusable for different environments.

**Review by kozlovic — CHANGES_REQUESTED** [review]
I am not knowledgeable on WASM to really comment, but there are conflicts that would need to be resolved before being able to merge.
For Go 1.17 support, please *add* (not replace) build directive as such:
```
//go:build <directive>
```
So suppose you want to have:
```
// +build wasm
```
it should become:
```
//go:build wasm
// +build wasm
```
etc.. Alternatively, if you are using Go 1.17, do a "go fmt" that should add those for you.

**Comment by gedw99** [comment]
+1 Any movements on this ??

**Comment by derekcollison** [comment]
Since the other PR landed with direct connect clients etc, @neilalexander do you think you could rebase on main and update copyright to 2022? Will see if we can merge for 2.9.

**Comment by neilalexander** [comment]
@derekcollison Absolutely, I’ll do it first thing in the morning!

**Review by kozlovic — APPROVED** [review]
LGTM

**Comment by unicomp23** [comment]
what's holding up the merge?

**Comment by derekcollison** [comment]
Should be ok to merge, will wait on @kozlovic

**Pull request merged**
